File: src/Python/303.RangeSumQueryImmutable.py

class NumArray:

    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        n = len(nums)
        self.arr = []
        if n == 0:
            return
        self.arr.append(nums[0])
        for i in range(1, n):
            self.arr.append(self.arr[i - 1] + nums[i])

    def sumRange(self, i, j):
        """
        :type i: int
        :type j: int
        :rtype: int
        """
        if i == 0:
            return self.arr[j]
        return self.arr[j] - self.arr[i - 1]
    
if __name__ == '__main__':
    obj = NumArray([-2, 0, 3, -5, 2, -1])
    print(obj.sumRange(0, 2))
    print(obj.sumRange(3, 5))
    print(obj.sumRange(0, 5))
else:
    pass

File: src/CSharp/301-400/303.RangeSumQueryImmutable.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.NumArray
{
public class NumArray {

    int[] arr;
    public NumArray(int[] nums) {
        int n = nums.Length;
        if (n == 0) {
            return;
        }
        arr = new int[n];
        arr[0] = nums[0];
        for (int i = 1; i < n; i++) {
            arr[i] = arr[i - 1] + nums[i];
        }
    }
    
    public int SumRange(int i, int j) {
        if (i == 0) {
            return arr[j];
        } else {
            return arr[j] - arr[i - 1];
        }
    }
}
}